frontend/src/pages/SignupPage.js

```diff
@@ -0,0 +1,113 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useState } from "react";
+import axios from "axios";
+import { URL_USER_SVC } from "../configs";
+import { STATUS_CODE_CONFLICT, STATUS_CODE_CREATED } from "../constants";
+import { Link } from "react-router-dom";
+
+function SignupPage() {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
I think it would be better if you put username and password together as an object.